benchmarks/layeredflow/layeredflow.cc

```diff
@@ -0,0 +1,461 @@
+    void
+    LayeredFlowMaterial<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      //create a global section in the parameter file for parameters
+      //that describe this benchmark. note that we declare them here
+      //in the material model, but other kinds of plugins (e.g., the gravity
+      //model below) may also read these parameters even though they do not
+      //declare them
+      prm.enter_subsection("LayeredFlow benchmark");
+      {
+        prm.declare_entry("Viscosity parameter beta", "1",
+                          Patterns::Double (),
+                          "Viscosity parameter beta in the LayeredFlow benchmark.");
+        prm.declare_entry("Viscosity parameter epsilon", "0.1",
+                          Patterns::Double (),
+                          "Viscosity parameter epsilon in the LayeredFlow benchmark.");
+      }
+      prm.leave_subsection();
+    }
+
+
+    template <int dim>
+    void
+    LayeredFlowMaterial<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("LayeredFlow benchmark");
+      {
+        beta = prm.get_double ("Viscosity parameter beta");
+        epsilon = prm.get_double ("Viscosity parameter epsilon");
+      }
+      prm.leave_subsection();
+
+      // Declare dependencies on solution variables
+      this->model_dependence.viscosity = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.density = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.compressibility = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.specific_heat = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.thermal_conductivity = MaterialModel::NonlinearDependence::none;
+    }
+
+
+    template <int dim>
+    double
+    LayeredFlowMaterial<dim>::get_beta() const
+    {
+      return beta;
+    }
+
+    template <int dim>
+    double
+    LayeredFlowMaterial<dim>::get_epsilon() const
+    {
+      return epsilon;
+    }
+
+    template <int dim>
+    LayeredFlowBoundary<dim>::LayeredFlowBoundary ()
+      :
+      beta (0),
+      epsilon (0)
+    {}
+
+
+
+    template <>
+    Tensor<1,2>
+    LayeredFlowBoundary<2>::
+    boundary_velocity (const types::boundary_id ,
+                       const Point<2> &p) const
+    {
+      const LayeredFlowMaterial<2> *
+      material_model
+        = dynamic_cast<const LayeredFlowMaterial<2> *>(&this->get_material_model());
+      return AnalyticSolutions::LayeredFlow_velocity (p, material_model->get_beta(),
+                                                      material_model->get_epsilon());
+    }
+
+
+
+    template <>
+    Tensor<1,3>
+    LayeredFlowBoundary<3>::
+    boundary_velocity (const types::boundary_id ,
+                       const Point<3> &) const
+    {
+      Assert (false, ExcNotImplemented());
+      return Tensor<1,3>();
+    }
+
+
+
+
+    /**
+     *gravity model for the LayeredFlow benchmark
+    */
```

Comment:
same as other file

Reply:
I know this is pre-existing. I've fixed this in #2511 now.

Reply:
oups